pkg/controller/history/controller_history.go

```diff
@@ -113,7 +113,9 @@ func HashControllerRevision(revision *apps.ControllerRevision, probe *int32) str
 
 // SortControllerRevisions sorts revisions by their Revision.
 func SortControllerRevisions(revisions []*apps.ControllerRevision) {
-	sort.Sort(byRevision(revisions))
+	// Stably sort first by name to break revision number ties by name
+	sort.Stable(byName(revisions))
+	sort.Stable(byRevision(revisions))
```

Comment:
You don't need to sort it twice. In the `Less` function of `byRevision`, use a tie-breaker there when two revisions are equal. I'd prefer using something like `CreationTimestamp` as a tie-breaker.

Reply:
We can use `Name` as another tie-breaker if both `Revision` and `CreationTimestamp` are equal.